src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,171 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Used to parse user input into recognised commands and execute them.
+ */
+public class Parser {
+
+    /**
+     * Parses the given user input (commandString) and executes the correct action based on that. Returns
+     * a boolean to indicate if the application should close due to the user's input.
+     *
+     * @param commandString string that contains the user's input command
+     * @param taskList a list that contains all tasks
+     * @return true if the application should exit. Otherwise, returns false.
+     */
+    public static String parseAndExecute(String commandString, TaskList taskList) {
+        Scanner stringStream = new Scanner(commandString);
+        String command = stringStream.next();
+
+        if (command.equalsIgnoreCase("bye")) {
+            return "bye";
+        } else if (command.equalsIgnoreCase("list")) {
+            return handleList(taskList);
+        } else if (command.equalsIgnoreCase("mark")) {
+            return handleMark(stringStream, taskList);
+        } else if (command.equalsIgnoreCase("unmark")) {
+            return handleUnmark(stringStream, taskList);
+        } else if (command.equalsIgnoreCase("deadline")) {
+            return handleDeadline(stringStream, taskList);
+        } else if (command.equalsIgnoreCase("event")) {
+            return handleEvent(stringStream, taskList);
+        } else if (command.equalsIgnoreCase("todo")) {
+            return handleToDo(stringStream, taskList);
+        } else if (command.equalsIgnoreCase("delete")) {
+            return handleDelete(stringStream, taskList);
+        } else if (command.equalsIgnoreCase("find")) {
+            return handleFind(stringStream, taskList);
+        } else {
+            return Ui.genUnknownCommandMsg();
+        }
+    }
+
+    private static String handleList(TaskList taskList) {
```

Comment:
For the following handle~ methods, maybe you can try to encapsulated them into a new packet called Command and create different types of commands there.